Format code with ESLint (#5, partially #6)

nitzel opened the pull request:
- Added ESLint
- Added curly braces after `if/else/for` if they were missing
- Consistently using `tab`-indentation now
- Added `spaces` between parameters, operators etc
- Replaced `var` with `let` and `const`
- Turned all `double quotes` into `single quotes`
- Used `string interpolation` instead of concatenation
- Removed superfluous escaping backslashes from regexes
- Added globally used variables to `.eslintrc.globals` (there are a lot!)
- Added semicolons
- Opening and closing parenthesis for a function's parameters are now on a line each or on the same line

This closes #5 and makes some progress on #6

nitzel committed:
Add ESLint for server.js #6

nitzel committed:
Apply code style to chat.js #6

nitzel committed:
Apply ESLint formatting to style_selector.js #6

nitzel committed:
Apply ESLint formatting to ptn.js #6

nitzel committed:
Apply ESLint formatting to board.js and main.js

nitzel committed:
Change indentation to tabs #6

nitzel committed:
Add curly braces to all if/else/for #5

nitzel committed:
Requier function parameters to be on a single line or a new line each  #5

nitzel commented on Static/js/chat.js:
```diff
@@ -1,265 +1,271 @@
+		$('#chat-toggle-button').css('right', this.chat_width + 5);
+		$('#chat-toggle-text').html('>><br>c<br>h<br>a<br>t');
+		$('#chat').removeClass('hidden');
+		if (fixedcamera || true) {
```
@NoHatCoder Do you know why this check exists here and in line 216? `|| true` looks like someone wanted to temporarily disable the condition - but if that's a permanent thing, we could remove the entire `if` statement.

NoHatCoder replied:
I wrote this while I was still unsure how all the cameras would work, I think we can just remove the if now.

NoHatCoder commented on Static/js/board.js:
```diff
@@ -1789,40 +1931,37 @@ var board = {
 		// but doesn't specify the details of the actual layout
-		var tpsRE = /\[(TPS\s*)?\"?\s*([,x12345678SC\/]+)(\s+([\d+]))?(\s+(\d+|-))?\s*\"?\s*\]/;
-		var result = tpsRE.exec(tps);
+		const tpsRE = /\[(TPS\s*)?"?\s*([,x12345678SC/]+)(\s+([\d+]))?(\s+(\d+|-))?\s*"?\s*\]/;
```
Could we maybe not get rid of "unnecessary" escapes? Seeing / unescaped in the middle of an expression gives me the creeps.

nitzel commented:
- [x] Remove all occurences of `// eslint-disable-next-line no-use-before-define` because it's disabled globally
- [x] Check if any other globally disabled rules still have related superfluous `// eslint-disable` comments
- [x] Remove `if(fixedcamera)` checks around `generateCamera()`
- [x] Escape `/` and `"` in regexes
- [x] ⛔  Enforce opening and closing parenthesis of `if` to be on their own lines or on the same line (like functions) 
  - I've had a look around [brace-style](https://eslint.org/docs/rules/brace-style), [operator-linebreak](https://eslint.org/docs/rules/operator-linebreak) and a few other but there's nothing like https://eslint.org/docs/rules/function-paren-newline for if instead of functions signatures.

nitzel committed:
Remove no longer require fixedcamera checks around generateCamera()

nitzel committed:
Escape " and / in regexes again

nitzel committed:
Fix weird linebreaks after variable declarations

nitzel committed:
Remove superfluous usages of eslint-disable-next-line no-use-before-define (since it's globally disabled)

nitzel committed:
Remove superfluous usage of eslint-disable-next-line no-cond-assign (as it's globally disabled)

nitzel committed:
Remove unused variables, turn on ESLint no-unused-vars for the local scope

Variables/functions exposed through .eslintrc.globals still show errors.
Also functions used in the html files aren't recognized as used by
ESLint.
Thus they need the // eslint-disable no-unused-vars line above them

nitzel committed:
Remove more unused variables from server.js (missed them in the last commit)

NoHatCoder committed:
Merge branch 'master' into feature/6-refactor-code
